src/browser/web_storage.rs

```diff
@@ -174,8 +191,14 @@ pub trait WebStorage {
     /// or serialize the value or insert/update the pair.
     ///
     /// [MDN reference](https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem)
-    fn insert<T: Serialize + ?Sized>(key: impl AsRef<str>, value: &T) -> Result<()> {
-        let value = serde_json::to_string(value).map_err(WebStorageError::SerdeError)?;
+    fn insert<S, T>(key: S, value: &T) -> Result<()>
+    where
+        S: AsRef<str>,
+        T: Serialize,
```

Comment:
Why are explicitly named typed parameters better than `impl`s? They allow to use turbofish but I don't know if it makes sense for this case.
Or I would at least replace `S` and `T` with ``K` and `V`.

Reply:
Overall I'd say `where` clauses are a bit clearer that `impl`s, but I realize that Seed uses `impl`s a lot, so if I'm just adding confusion to an established pattern, I'm okay with reverting.

Is there a difference performance-wise, or is that only when `dyn` is present?

Reply:
> Is there a difference performance-wise, or is that only when dyn is present?

I don't think so. I use mostly `impl` instead of "classic" generics / `where` because I don't need to come up with names and it feels a bit more natural to read to me.
The only difference I know - when you want to write a function similar to `Iterator`'s `collect` and call it with turbofish (e.g. `.collect::<Vec<_>>()`), then you have to use generics, turbofish + impl Trait causes compilation error.

`dyn` is used in Seed to simplify some types or when it's the only option. Also it may reduce a wasm file size a bit so it isn't a problem in most cases.

Reply:
Then I guess it's just our difference styles, since we both consider our own way to be easier to read :rofl: 

How about I at least change the type param names, as you suggest. Then going forward in future PRs I'll try to leave the existing style as-is.